### PR TITLE
[FONT][FREETYPE] Fix IntRequestFontSize function to render correctly in font size (CORE-11620, CORE-13840)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This Pull Request fixes font rendering in font size.
JIRA issue: [CORE-11620](https://jira.reactos.org/browse/CORE-11620), [CORE-13840](https://jira.reactos.org/browse/CORE-13840)

### CODE REVIEW

**Comment by JoachimHenze** [comment]
I applied your PR manually and I **love** the effect on Total Commander 8.52a:
pr330

But the changed text size in rapps details section makes me wonder, as it looks now different than when executed on Windows...
rapps_pr330
Either some other bug was unhidden here or the patch may not be entirely correct?

**Comment by katahiromz** [comment]
I adjusted the stock font sizes to win2k3 compatible.

**Comment by JoachimHenze** [comment]
After commit 2 I really like this PR **from a pure testers point of view**. I could not find any unexpected text-rendering-glitches in my tests anymore. If others are happy with semantic code-changes as well, then let's get that in.

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -3002,6 +2998,17 @@ IntRequestFontSize(PDC dc, FT_Face face, LONG Width, LONG Height)
...
 
+    if (Height < 0)
+    {
+        /* case (A): lfHeight is negative: the character height */
+        Height = -Height;
+    }
+    else if (Height > 0)
+    {
+        /* case (B): lfHeight is positive: the cell height */
+        Height = Height * face->units_per_EM / face->height;
```
Comment (JoachimHenze) [inline win32ss/gdi/ntgdi/freetype.c]: If Height==0 and face->height==0 wouldn't that crash? Or if face==NULL?

**Comment by yagoulas** [comment]
I wonder if there is a way to test the size of the stock fonts

File: win32ss/gdi/ntgdi/freetype.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: If face is NULL, FT_New_Memory_Face will fail. I think it doesn't come here in the case.
I can't deny the case that face->height can be zero.
I will add ASSERT(face != NULL); and ASSERT(face->height != 0);.
And if face->height == 0, then I will use height value 1.

**Comment by katahiromz** [comment]
@yagoulas: See the test program StockFont.zip of CORE-11620.

File: win32ss/gdi/ntgdi/freetype.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: Done.

File: win32ss/gdi/ntgdi/freetype.c [hunk 2]
```diff
@@ -3003,7 +3005,7 @@ IntRequestFontSize(PDC dc, FT_Face face, LONG Width, LONG Height)
         /* case (A): lfHeight is negative: the character height */
         Height = -Height;
     }
-    else if (Height > 0)
+    else if (Height > 0 && face->height > 0)
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/freetype.c]: Do we cover everything now? aka: Height > 0 but face->height <= 0 (if it's possible at all)?

**Comment by HBelusca** [comment]
Someone please check whether it fixes the console properties font display :)

File: win32ss/gdi/ntgdi/freetype.c (hunk 2 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: face->height <= 0 means an illegal font.

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: face->height <= 0 means an illegal font.

Comment (HBelusca) [reply win32ss/gdi/ntgdi/freetype.c]: So this is handled before/elsewhere?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: I cannot consider the height when the font was invalid. What to do?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: I added "ASSERT(face->height > 0);".

**Comment by katahiromz** [comment]
This pull request is too old. I will retry PR.

**Comment by katahiromz** [comment]
Retried at #410.

**Pull request closed without merging**